book/src/simple_application/drawing.md

```diff
@@ -0,0 +1,172 @@
+    .with(square.clone())
+    .with(ball)
+    .with(LocalTransform::default())
+    .with(Transform::default())
+    .build();
+```
+Some of the components we can see here are built into amethyst to control how
+things are drawn, and we can add an unlimited number of our own components.
+We'll see how some of the built-in components work later.
+
+# Creating a ball entity
+
+We're going to spend the rest of this stage learning how to draw to the screen.
+We want to create an entity, and attach the required component information to
+it so amethyst can draw it.
+
+Let's look at the second parameter to `on_start`, a `&mut AssetManager`. The
+asset manager is responsible for holding rendering information, including
+meshes and textures. We're going to tell it to make space for our meshes and
+textures, then to store some very simple ones that we need to draw our ball. We
+will use a flat color for our texture, so there will be no need to load any
+image files, and we will use a simple square for our ball, containing only 2
+triangles.
+
+## Adding a renderable
+
+```rust
+
+// Generate a square mesh
+assets.register_asset::<Mesh>();
+assets.register_asset::<Texture>();
+assets.load_asset_from_data::<Texture, [f32; 4]>("white", [1.0, 1.0, 1.0, 1.0]);
+let square_verts = gen_rectangle(1.0, 1.0);
+assets.load_asset_from_data::<Mesh, Vec<VertexPosNormal>>("square", square_verts);
+let square = assets.create_renderable("square", "white", "white", "white", 1.0).unwrap();
+```
+
+The stages are
+ 1. Register asset types, so we can store assets of that type
+ 2. Load a single pixel texture into our asset manager. The texture is 4 `f32`s
+    wide, one for each of rgba. It is labelled with the string "white".
+ 3. Generate vertices, normals and texture coordinates for a square. This is
+    not partiularly specific to amethyst, so I'm not going to explain it all in
+    detail, but very briefly there are 2 triangles of 3 vertices each. Normals
+    all point up, and texture triangles are over our single pixel. The asset is
+    labelled "square".
+ 4. Finally, we create a *renderable* out of the mesh and texture. This is
+    something we can draw, essentially just a block of data. We will use this 
+    to draw all the objects in our scene (they will all be white rectangles. 
+    We refer to assets by their string label. See the method signature in the 
+    docs for what the parameters are.
+
+## Creating the pipeline
+
+In amethyst (and probably elsewhere, I'm not an expert), the pipeline detials
+the number and order of the rendering passes. We will have a single pass,
+consisting of a layer to clear the previous frame, and a layer to draw our ball 
+(and paddles, added later).
+
+```rust
+let layer = Layer::new("main",
+                       vec![Clear::new([0.0, 0.0, 0.0, 1.0]),
+                            DrawFlat::new("main", "main")]);
+pipe.layers.push(layer);
+```
+
+We use the `DrawFlat` program because we don't need to apply shading (we're
```

Comment:
should we say program or shader? Both are technically correct, but I feel like program implies something that a user runs